src/lib/core/BUILD.gn

```diff
@@ -84,6 +84,12 @@ source_set("chip_config_header") {
   allow_circular_includes_from = [ "${chip_root}/src/ble:ble_config_header" ]
 }
 
+source_set("chip_error") {
+  sources = [ "CHIPError.h" ]
```

Comment:
ChipError has a cpp. I don't think they can be separated as CPP implements error needs.

Reply:
If we do that, CHIPError.cpp depends on ErrorStr.h.  Which depends on CHIPError.h.  Which means that all three files should be a single source_set, yes?  Which sort of reinforces the TODO bit above....

Also note that CHIPError.h calls FormatError, which is provided by ErrorStr.cpp....

Reply:
Created https://github.com/project-chip/connectedhomeip/pull/29494 to do the CHIPError bits, then will do the rest of this PR on top of those.